modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1215,7 +1215,12 @@ void GDScriptAnalyzer::resolve_constant(GDScriptParser::ConstantNode *p_constant
 void GDScriptAnalyzer::resolve_assert(GDScriptParser::AssertNode *p_assert) {
 	reduce_expression(p_assert->condition);
 	if (p_assert->message != nullptr) {
-		reduce_literal(p_assert->message);
+		reduce_expression(p_assert->message);
+		if (!p_assert->message->is_constant) {
+			push_error(R"(Expected constant string for assert error message.)", p_assert->message);
+		} else if (p_assert->message->reduced_value.get_type() != Variant::STRING) {
+			push_error(R"(Expected constant string for assert error message.)", p_assert->message);
```

Comment:
Aren't these blocks identical?

```suggestion
		if (!p_assert->message->is_constant || p_assert->message->reduced_value.get_type() != Variant::STRING) {
			push_error(R"(Expected constant string for assert error message.)", p_assert->message);
```

Reply:
yeah, I'll change it. It was different error messages for resolving enum values  1. should be constant, 2. should be an integer. here I combined the messages as "expected constant string"